### PR TITLE
Add unit test for download, encodings, hashing, and others

### PR DESCRIPTION
**Pull request by karan6181**
## Description of changes:
- Added unittest for different cloud blob storage download APIs.
- Added unittest for MDS, XSV, and JSON encodings
- Added unittest for supported hashing algorithms
- Added unittest for util function
- Improved unittest for compression algorithm

## Merge Checklist:
_Put an `x` without space in the boxes that apply. If you are unsure about any checklist, please don't hesitate to ask. We are here to help! This is simply a reminder of what we are going to look for before merging your pull request._

### General
- [x] I have read the [contributor guidelines](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md)
- [ ] This is a documentation change or typo fix. If so, skip the rest of this checklist.
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the MosaicML team.
- [x] I have updated any necessary documentation, including [README](https://github.com/mosaicml/streaming/blob/main/README.md) and [API docs](https://github.com/mosaicml/streaming/tree/main/docs) (if appropriate).

### Tests
- [x] I ran `pre-commit` on my change. (check out the `pre-commit` section of [prerequisites](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#prerequisites))
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate).
- [x] I ran the tests locally to make sure it pass. (check out [testing](https://github.com/mosaicml/streaming/blob/main/CONTRIBUTING.md#running-tests))
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes.

### CODE REVIEW

File: tests/test_encodings.py [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+        with pytest.raises(AttributeError):
+            int_enc = mdsEnc.Int()
+            _ = int_enc.encode(data)
+
+    @pytest.mark.parametrize('mode', ['I', 'L', 'RGB'])
+    def test_pil_encode_decode(self, mode: str):
+        pil_enc = mdsEnc.PIL()
+        assert pil_enc.size is None
+
+        # Creating the 32 X 32 NumPy Array with random values
```
Comment (knighton) [inline tests/test_encodings.py:82]: maybe conventionally a lowercase x here and elsewhere?

File: streaming/base/format/xsv/encodings.py [hunk 2]
```diff
@@ -94,28 +100,28 @@ def is_xsv_encoding(encoding: str) -> bool:
...
 
 def xsv_encode(encoding: str, value: Any) -> str:
-    """Encode the object.
+    """Encode the given data from the original object to string.
 
     Args:
-        encoding (str): The encoding.
-        value (Any): The object.
+        encoding (str): Encoding.
+        value (Any): Decoded object.
```
Comment (knighton) [inline streaming/base/format/xsv/encodings.py:107]: Encoded

Comment (knighton) [reply streaming/base/format/xsv/encodings.py:107]: *Object to encode

**Review by knighton — APPROVED** [review]

File: tests/test_encodings.py (hunk 1 above)
Comment (karan6181) [reply tests/test_encodings.py]: Done in https://github.com/mosaicml/streaming/pull/72/commits/f202a4ff6281c757fc5cc6f0d11e3e2edfbef424.

File: streaming/base/format/xsv/encodings.py (hunk 2 above)
Comment (karan6181) [reply streaming/base/format/xsv/encodings.py]: Done in https://github.com/mosaicml/streaming/pull/72/commits/f202a4ff6281c757fc5cc6f0d11e3e2edfbef424.

**Pull request merged**